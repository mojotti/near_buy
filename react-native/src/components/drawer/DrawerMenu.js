import React, { Component } from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { logout } from '../../redux/actions/AuthorizationAction';
import { styles } from '../../static/styles/DrawerStyles';
import MenuItem from './MenuItem';

class DrawerMenu extends Component {
  render() {
    const allItemsProps = {
      itemName: 'All items',
      iconName: 'public',
      iconSize: 27,
      navigationRoute: 'ItemExplorer',
      navigation: this.props.navigation,
    };
    const myItemProps = {
      itemName: 'My items',
      iconName: 'account-circle',
      iconSize: 27,
      navigationRoute: 'UserItems',
      navigation: this.props.navigation,
    };
    const myChatProps = {
      itemName: 'Chats',
      iconName: 'comment',
      iconSize: 27,
      navigationRoute: 'MyChats',
      navigation: this.props.navigation,
    };

    return (
      <View style={styles.container}>
        <View>
          <MenuItem {...allItemsProps} />
          <MenuItem {...myItemProps} />
          <MenuItem {...myChatProps} />
        </View>

        <TouchableHighlight
          onPress={() => this.props.onLogout()}
          style={styles.logoutButton}
        >
          <Text style={styles.loginText}>Logout</Text>
        </TouchableHighlight>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onLogout: () => {
      dispatch(logout());
    },
  };
};

DrawerMenu.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  onLogout: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(DrawerMenu);
