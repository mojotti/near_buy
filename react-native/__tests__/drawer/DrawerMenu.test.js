import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import DrawerMenu from '../../src/components/drawer/DrawerMenu';

const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {};
const store = mockStore(initialState);


describe('<DrawerMenu />', () => {
  test('renders correctly', () => {
    const navi = { navigate: () => {} }
    const wrapper = shallow(<DrawerMenu store={store} navigation={navi}/>);
    expect(wrapper.dive()).toMatchSnapshot();
  });

  test('dispatches correct action on logout', () => {
    const navi = { navigate: () => {} }
    const wrapper = shallow(<DrawerMenu store={store} navigation={navi}/>);

    const logoutButton = wrapper.dive().find('TouchableHighlight');
    logoutButton.simulate('Press');

    const expectedActions = [{ type: 'LOGOUT' }];
    expect(store.getActions()).toEqual(expectedActions);
  });
});
