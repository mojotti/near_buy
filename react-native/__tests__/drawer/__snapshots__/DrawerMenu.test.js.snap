// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerMenu /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
      "paddingTop": 100,
    }
  }
>
  <View>
    <Connect(_MenuItem)
      iconName="public"
      iconSize={27}
      itemName="All items"
      navigation={
        Object {
          "navigate": [Function],
        }
      }
      navigationRoute="ItemExplorer"
    />
    <Connect(_MenuItem)
      iconName="account-circle"
      iconSize={27}
      itemName="My items"
      navigation={
        Object {
          "navigate": [Function],
        }
      }
      navigationRoute="UserItems"
    />
    <Connect(_MenuItem)
      iconName="comment"
      iconSize={27}
      itemName="Chats"
      navigation={
        Object {
          "navigate": [Function],
        }
      }
      navigationRoute="MyChats"
    />
  </View>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#4d4dff",
        "borderRadius": 30,
        "elevation": 1,
        "marginBottom": 30,
        "marginVertical": 10,
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 397.5,
      }
    }
    underlayColor="black"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": null,
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Logout
    </Text>
  </TouchableHighlight>
</View>
`;
